src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -704,6 +724,11 @@ AdaptiveExecutor(CitusScanState *scanState)
 
 	FinishDistributedExecution(execution);
 
+	if (hasDependentJobs)
+	{
+		DoRepartitionCleanup(jobIdList, true);
```

Comment:
Why is it ok to error here? We still want to make sure we clean up as much as possible.

Reply:
Yeah, that is true.